Makefile

```diff
@@ -85,6 +84,12 @@ vet:
 # Generate code & docs
 generate: install-tools api-reference
 	controller-gen object:headerFile="hack/NOTICE.go.txt" paths="./..."
+
+generate-client-set:
+	# This one just needs to be fetched and not installed, get & mod so it ends up in the right place.
+	# Note we grep it out above, and just do a go get & go mod for it.
+	go get -d k8s.io/code-generator
+	go mod vendor
```

Comment:
Are we sticking with vendoring then?

Reply:
Only when you are generating the code. The code generation script `./hack/update-codegen.sh` refers to ` ./vendor/k8s.io/code-generator` at the moment. After running the the make target, `vendor/` can be safely deleted and you don't need to check it in.

Reply:
I feel a bit weird having a package-level structure (i.e. `vendor/`) existing just for this one script, even if it's temporary for this one script.

Having said that, it's in the gitignore and you didn't introduce it in this PR, so feel free to go ahead with this change regardless.